cmd/kubelet/kubelet.go

```diff
@@ -99,6 +109,28 @@ func main() {
 		die(err)
 	}
 
+	// If the config returned from the controller is a different object than defaultConfig,
+	// we must enforce flag precedence by re-parsing the command line into the new object.
+	// This is necessary to preserve backwards-compatibility across binary upgrades.
+	// See issue #56171 for more details.
+	if kubeletConfig != defaultConfig {
```

Comment:
Why do we need to compare to default config? Seems like we'd always want to reparse over the config

Reply:
defaultConfig consists of the built-in defaults, with flags already parsed over it. There shouldn't really be a reason to parse the same flags over it again.

Reply:
OTOH, we don't really need this optimization, so I stripped it for simplicity. Since the parse accumulation issue is related to re-using a flagset, not re-using an object, that's not a concern here either.